src/saveload/order_sl.cpp

```diff
@@ -158,50 +184,62 @@ struct ORDRChunkHandler : ChunkHandler {
 				SlCopy(&orders[0], len, SLE_UINT32);
 
 				for (size_t i = 0; i < len; ++i) {
-					new (OrderID(static_cast<uint32_t>(i))) Order(GB(orders[i], 0, 8), GB(orders[i], 8, 8), GB(orders[i], 16, 16));
+					auto &item = AllocateOldOrder(i);
+					item.order = Order(GB(orders[i], 0, 8), GB(orders[i], 8, 8), GB(orders[i], 16, 16));
 				}
 			}
 
-			/* Update all the next pointer */
-			for (Order *o : Order::Iterate()) {
-				size_t order_index = o->index.base();
-				/* Delete invalid orders */
-				if (o->IsType(OT_NOTHING)) {
-					delete o;
-					continue;
+			/* Update all the next pointer. The orders were built like this:
+			 * While the order is valid, set the previous will get its next pointer set */
```

Comment:
I don't quite understand this sentence. Might the first `set` need to be removed?

Reply:
Good point, I just re-indented the comment, not my wording :)